tanner/emulators/base.py

```diff
@@ -23,57 +15,66 @@ def __init__(self, base_dir, db_name, loop=None):
             'sqli': sqli.SqliEmulator(db_name, base_dir),
             'cmd_exec': cmd_exec.CmdExecEmulator()
         }
+        self.get_emulators = ['sqli', 'rfi', 'lfi', 'xss', 'cmd_exec']
+        self.post_emulators = ['sqli', 'rfi', 'lfi', 'xss', 'cmd_exec']
 
-    async def handle_post(self, session, data):
+    def extract_get_data(self, path):
+        """
+        Return all the GET parameter
+        :param path: The URL path from which GET parameters are to be extracted
+        :return: A MultiDictProxy object containg name and value of parameters
+        """
+        path = urllib.parse.unquote(path)
+        encodings = [('&&', '%26%26'), (';', '%3B')] 
+        for value, encoded_value in encodings:
+            path = path.replace(value, encoded_value)
+        get_data = yarl.URL(path).query
+        return get_data
+
+    async def get_emulation_result(self, session, data, target_emulators):
+        """
+        Return emulation result for the vulnerabilty of highest order
+        :param session: Current active session
```

Comment:
Yes, you are right. I don't want to remove docstrings. I completely understand your point of view, I read lots of source code and know this feeling. But when I read the new code, the most important thing for me is input/output type. I saw the string "Current active session" and asked myself: "is it string, is it number or is it other type?". But if I see the string "Session class object" it makes difference. I don't need to find the place where the current session is created, to understand it's type. I 100% sure that I'll find all the information in session class.